src/itemdisplay.zig

```diff
@@ -0,0 +1,141 @@
+const std = @import("std");
+
+const blocks = @import("blocks.zig");
+const chunk_zig = @import("chunk.zig");
+const ServerChunk = chunk_zig.ServerChunk;
+const game = @import("game.zig");
+const Player = game.Player;
+const graphics = @import("graphics.zig");
+const c = graphics.c;
+const main = @import("main.zig");
+const utils = @import("utils.zig");
+const vec = @import("vec.zig");
+const Mat4f = vec.Mat4f;
+const Vec3d = vec.Vec3d;
+const Vec3f = vec.Vec3f;
+const Vec3i = vec.Vec3i;
+const NeverFailingAllocator = main.utils.NeverFailingAllocator;
+const ItemModelStore = @import("itemdrop.zig").ItemModelStore;
+
+pub var showItem: bool = true;
+
+// Going to handle item animations and other things like - bobbing, interpolation
+pub const PlayerItemDisplayManager = struct {
+	pub var cameraFollow: Vec3f = .{0, 0, 0};
+	var lastTime: i16 = 0;
+	var timeDifference: utils.TimeDifference = .{};
+
+	pub fn init() void {
+		cameraFollow = game.camera.rotation;
+	}
+
+	pub fn update() void {
+		var time = @as(i16, @truncate(std.time.milliTimestamp()));
+		time -%= timeDifference.difference.load(.monotonic);
+		const deltaTime = @as(f32, @floatFromInt(time -% lastTime))/1000;
+
+		const blend: f32 = deltaTime * 19;
+		cameraFollow = vec.lerp(cameraFollow, game.camera.rotation, blend);
+		lastTime = time;
+	}
+};
+
+pub const PlayerItemDisplay = struct { // MARK: PlayerItemDisplay
+	// stolen from itemdrop.zig with love!!!
+	var itemShader: graphics.Shader = undefined;
+	var itemUniforms: struct {
+		projectionMatrix: c_int,
+		modelMatrix: c_int,
+		viewMatrix: c_int,
+		modelPosition: c_int,
+		ambientLight: c_int,
+		modelIndex: c_int,
+		block: c_int,
+		time: c_int,
+		texture_sampler: c_int,
+		emissionSampler: c_int,
+		reflectivityAndAbsorptionSampler: c_int,
+		reflectionMap: c_int,
+		reflectionMapSize: c_int,
+		contrast: c_int,
+	} = undefined;
+
+	pub fn init() void {
+		itemShader = graphics.Shader.initAndGetUniforms("assets/cubyz/shaders/item_drop.vs", "assets/cubyz/shaders/item_drop.fs", "", &itemUniforms);
```

Comment:
Please use the shader and uniforms from the ItemDropRenderer.
Keeping the same uniforms in two different places just makes it needlessly difficult to change parameters.